LuaMenu/widgets/chobby/components/aioptions_window.lua

```diff
@@ -0,0 +1,282 @@
+AiOptionsWindow = ListWindow:extends{}
+
+function AiOptionsWindow:init(displayName, optionsPath, successFunc)
+	self:super('init', lobbyInterfaceHolder, displayName.." Options", false, "main_window", nil, {6, 7, 7, 4})
+	self.window:SetPos(nil, nil, 650, 700)
+	WG.Chobby.PriorityPopup(self.window, nil, nil, nil, true)
+
+	self.aioptions = {}
+
+	local applyBtn = Button:New {
+		right = 120,
+		y = 7,
+		width = 80,
+		height = 45,
+		caption = i18n("apply"),
+		font = WG.Chobby.Configuration:GetFont(3),
+		parent = self.window,
+		classname = "action_button",
+		OnClick = {
+			function()
+				successFunc(self.side, self.aioptions)
+				self:HideWindow()
+			end
+		},
+	}
+
+	-- AIOptions
+	local options = VFS.Include(optionsPath)
+	for i = #options, 1, -1 do
+		self:AddEntry(options[i], i)
+	end
+
+	-- AI side
+	local singleplayer = WG.Chobby.Configuration.singleplayer
+	if singleplayer then
+		local items = {}
+		for i, side in pairs(singleplayer.sidedata) do
+			items[i] = side.name
+		end
+
+		local data = {
+			name = "Select side",
```

Comment:
I don't think Chobby can display side so I'm not sure we should be enabling this for AIs only.
If we do add side support, players should be able to choose side as well.

Alternatively both players and AIs will need to choose the side ingame

Reply:
I'll remove Side from aioptions window. But there seems no alternative to setting AI's side from lobby.
I mean it can't be comfortable for a player running FFA with 10 AIs to waste more time after game starts on configuring more AI params, i think it had to be done during AddAI phase.

Also the way it's PRed now Side can't be random, and it should be thought of too. (MetalFaction's approach with 5th "Random" faction is not cute).

Reply:
I think it's fine to have side support if we display it correctly for both players and AI.
The only reason this doesn't exist is because ZK doesn't have sides